base/pair.jl

```diff
@@ -41,7 +41,12 @@ eltype(p::Type{Pair{A, B}}) where {A, B} = Union{A, B}
 iterate(p::Pair, i=1) = i > 2 ? nothing : (getfield(p, i), i + 1)
 indexed_iterate(p::Pair, i::Int, state=1) = (getfield(p, i), i + 1)
 
-hash(p::Pair, h::UInt) = hash(p.second, hash(p.first, h))
+const _pairhash_seed = if UInt === UInt64
+    0x94cb2bb20a28ce96
+else
+    0x1f60a087
+end::UInt
+hash(p::Pair, h::UInt) = hash(p.second, hash(p.first, hash(_pairhash_seed, h)))
```

Comment:
FWIW, this also passes the new tests while being simpler/cheaper:
```suggestion
hash(p::Pair, h::UInt) = hash(p.second, hash(p.first, ~h))
```
I think it relies on `hash(p.first, ~h) != ~hash(p.first, h)`. Not sure whether it has any other disadvantages.

Reply:
The problem with this suggestion is that another collection, unrelated to `Pair`, could define a `hash` method with the same body, resulting in unnecessary collisions between `Pair` values and values of this other collection. That's why I used a random seed.

Reply:
I see, makes sense. The `xor` is probably a good compromise, then.